### ISSUE TITLE
Invisible window after maximizing

### ISSUE THREAD

**Issue by pmoris** [wez/wezterm#6187]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

KWin/Wayland - kwin 6.1.5
plasmashell 6.1.5
Fedora 40

### WezTerm version

20240922_151228_2b76c63b

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Maximizing a wezterm window (via keyboard shortcut or title bar button) makes it disappear.

- It is still interactable: it blocks clicks through it and I can exit the shell via ctrl+D if the window is focused (as seen on my taskbar).
- It shows up with a functional thumbnail preview in my task bar and in alt-tab preview (only in the small preview, not a rendered full screen desktop preview).
- I can tile the window and move it to different desktops, but it remains invisible. (I can see the tiling behaviour via the [KDE pager widget](https://userbase.kde.org/Plasma/Pager)).

I see the same behaviour on both my laptop or when using an external monitor. I observed a similar problem a while ago when using vscode, but not now. Kwin debug tells me wezterm is running in wayland mode, while vscode is not, but other wayland windows do not show the same issue (e.g., firefox). Restarting plasmashell does not fix the problem.

Potentially related problem: https://gitlab.gnome.org/GNOME/mutter/-/issues/3435 

Wayland logs do not show anything useful: 

- Loads of errors like `Sep 24 11:26:52 fedora-itm kwin_wayland[2714]: kwin_scene_opengl: 0x2: GL_INVALID_OPERATION in glDrawBuffers(unsupported buffer GL_BACK_LEFT)` (all the time).
- After maximizing a window, the following error shows up `Sep 24 11:30:57 fedora-itm kwin_wayland[2714]: This plugin does not support raise()` (but it also appears when simply alt-tabbing, so it does not seem to be related).

Lastly, I've attached  the full wezterm log. Since it grows so quickly, I waited a bit between starting a new session (`WEZTERM_LOG=debug wezterm start --always-new-process 2> wezterm-maximize.log` ~ 11:36:13) and maximizing it (~11:36:33). I've tried to extract the relevant part in the section below too.

[wezterm-maximize.log](https://github.com/user-attachments/files/17112153/wezterm-maximize.log)

### To Reproduce

Open a wezterm window and press the maximize button.

### Configuration

Problem persists with or without using a config (currently only sets a theme).

### Expected Behavior

Window should maximize and stay visible.

### Logs

```
11:36:14.034  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.556  DEBUG  window::os::wayland::window            > Config: self.window_state=None, states: MAXIMIZED WindowState(MAXIMIZED | ACTIVATED)
11:36:33.556  DEBUG  window::os::wayland::window            > dispatch_pending_event self.window_state=(empty), pending:MAXIMIZED
11:36:33.556  DEBUG  wezterm_gui::termwindow                > Resized { dimensions: Dimensions { pixel_width: 2557, pixel_height: 1376, dpi: 96 }, window_state: MAXIMIZED, live_resizing: false }
11:36:33.556  DEBUG  wezterm_term::screen                   > resize screen to 253x60 dpi=96
11:36:33.557  DEBUG  wezterm_term::screen                   > resize screen to 253x60 dpi=96
11:36:33.557  DEBUG  wezterm_gui::termwindow                > NeedRepaint
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > cluster_resolver: ClusterResolver {
    map: {
        1: ClusterInfo {
            start: 1,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
        0: ClusterInfo {
            start: 0,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
        6: ClusterInfo {
            start: 6,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
        2: ClusterInfo {
            start: 2,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
        4: ClusterInfo {
            start: 4,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
        5: ClusterInfo {
            start: 5,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
        3: ClusterInfo {
            start: 3,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
    },
    presentation_width: None,
    start_by_cell_idx: {},
}
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 0 -> Info { cluster: 0, len: 1, codepoint: 21, x_advance: 576, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 1 -> Info { cluster: 1, len: 1, codepoint: 30, x_advance: 320, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 2 -> Info { cluster: 2, len: 1, codepoint: 4, x_advance: 256, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 3 -> Info { cluster: 3, len: 1, codepoint: 70, x_advance: 576, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 4 -> Info { cluster: 4, len: 1, codepoint: 69, x_advance: 576, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 5 -> Info { cluster: 5, len: 1, codepoint: 87, x_advance: 512, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.559  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 6 -> Info { cluster: 6, len: 1, codepoint: 76, x_advance: 576, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.560  DEBUG  wezterm_font::shaper::harfbuzz         > font_idx=0 info_clusters: [
    [
        Info {
            cluster: 0,
            len: 1,
            codepoint: 21,
            x_advance: 576,
            y_advance: 0,
            x_offset: 0,
            y_offset: 0,
        },
    ],
    [
        Info {
            cluster: 1,
            len: 1,
            codepoint: 30,
            x_advance: 320,
            y_advance: 0,
            x_offset: 0,
            y_offset: 0,
        },
    ],
    [
        Info {
            cluster: 2,
            len: 1,
            codepoint: 4,
            x_advance: 256,
            y_advance: 0,
            x_offset: 0,
            y_offset: 0,
        },
    ],
    [
        Info {
            cluster: 3,
            len: 1,
            codepoint: 70,
            x_advance: 576,
            y_advance: 0,
            x_offset: 0,
            y_offset: 0,
        },
    ],
    [
        Info {
            cluster: 5,
            len: 1,
            codepoint: 87,
            x_advance: 512,
            y_advance: 0,
            x_offset: 0,
            y_offset: 0,
        },
    ],
    [
        Info {
            cluster: 6,
            len: 1,
            codepoint: 76,
            x_advance: 576,
            y_advance: 0,
            x_offset: 0,
            y_offset: 0,
        },
    ],
]
11:36:33.560  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=2.118065ms
11:36:33.560  DEBUG  wezterm_term::terminalstate::performer > perform Control(CarriageReturn)
11:36:33.560  DEBUG  wezterm_term::terminalstate::performer > perform CSI(Edit(EraseInLine(EraseToEndOfLine)))
11:36:33.561  DEBUG  wezterm_term::terminalstate::performer > perform Control(CarriageReturn)
11:36:33.561  DEBUG  wezterm_term::terminalstate::performer > perform CSI(Sgr(Foreground(PaletteIndex(5))))
11:36:33.561  DEBUG  wezterm_term::terminalstate            > Foreground(PaletteIndex(5))
11:36:33.561  DEBUG  wezterm_term::terminalstate::performer > perform PrintString("└─▶ ")
11:36:33.561  DEBUG  wezterm_term::terminalstate::performer > perform CSI(Sgr(Reset))
11:36:33.561  DEBUG  wezterm_term::terminalstate            > Reset
11:36:33.564  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=6.274953ms, fps=0.48625845
11:36:33.564  DEBUG  window::os::wayland::window            > Had configured an event
11:36:33.564  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.564  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.564  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.565  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.565  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.565  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.565  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:33.580  DEBUG  wezterm_gui::termwindow                > NeedRepaint
11:36:33.583  DEBUG  wezterm_font::shaper::harfbuzz         > cluster_resolver: ClusterResolver {
    map: {
        159: ClusterInfo {
            start: 159,
            byte_len: 1,
            cell_width: 1,
            incomplete: false,
        },
//
//
//
```
Cluster info keeps going for a bit, then
```
11:36:33.584  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 0 -> Info { cluster: 0, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.584  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 1 -> Info { cluster: 1, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.584  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 2 -> Info { cluster: 2, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.584  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 3 -> Info { cluster: 3, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.584  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 4 -> Info { cluster: 4, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.584  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 5 -> Info { cluster: 5, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.585  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 6 -> Info { cluster: 6, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.585  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 7 -> Info { cluster: 7, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.585  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 8 -> Info { cluster: 8, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
11:36:33.585  DEBUG  wezterm_font::shaper::harfbuzz         > hb info.cluster 9 -> Info { cluster: 9, len: 1, codepoint: 958, x_advance: 640, y_advance: 0, x_offset: 0, y_offset: 0 }
//
//
//
```
all the way until cluster 248, followed by another info_clusters section.

Eventually:

```
11:36:33.601  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=20.033513ms
11:36:33.602  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=21.570721ms, fps=0.48625845
11:36:34.514  DEBUG  wezterm_gui::termwindow                > RawKeyEvent(RawKeyEvent { key: Physical(LeftControl), modifiers: NONE, leds: (empty), phys_code: Some(LeftControl), raw_code: 37, repeat_count: 1, key_is_down: true, handled: Handled(false) })
11:36:34.515  DEBUG  wezterm_gui::termwindow                > KeyEvent(KeyEvent { key: LeftControl, modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(LeftControl), modifiers: NONE, leds: (empty), phys_code: Some(LeftControl), raw_code: 37, repeat_count: 1, key_is_down: true, handled: Handled(false) }) })
11:36:34.515  DEBUG  wezterm_gui::termwindow                > AdviseModifiersLedStatus(CTRL, (empty))
11:36:34.516  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:34.516  DEBUG  wezterm_gui::termwindow                > NeedRepaint
11:36:34.517  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=458.564µs
11:36:34.517  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=754.077µs, fps=0.48625845
11:36:34.517  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:34.518  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:34.558  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:34.558  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:34.559  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
11:36:34.940  DEBUG  wezterm_gui::termwindow                > RawKeyEvent(RawKeyEvent { key: Physical(D), modifiers: CTRL, leds: (empty), phys_code: Some(D), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) })
11:36:34.941  DEBUG  wezterm_gui::termwindow                > KeyEvent(KeyEvent { key: Char('d'), modifiers: CTRL, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(D), modifiers: CTRL, leds: (empty), phys_code: Some(D), raw_code: 40, repeat_count: 1, key_is_down: true, handled: Handled(false) }) })
11:36:34.941  ERROR  window::os::wayland::window            > hide_cursor: Missing enter event serial
11:36:34.942  DEBUG  wezterm_gui::termwindow                > NeedRepaint
11:36:34.942  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=526.194µs
11:36:34.943  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=854.378µs, fps=2.1674705
11:36:34.943  DEBUG  mux                                    > removing pane 0
11:36:34.944  DEBUG  mux                                    > killing pane 0
11:36:34.944  DEBUG  mux::localpane                         > killing process in pane 0, state is Running { child_waiter: Receiver { .. }, pid: Some(67247), signaller: ProcessSignaller { pid: Some(67247) }, killed: false }
11:36:34.945  DEBUG  mux::localpane                         > child terminated, new state is Dead
11:36:34.945  DEBUG  mux                                    > remove_tab_internal tab 0
11:36:34.945  DEBUG  mux                                    > panes to remove: []
11:36:34.945  DEBUG  mux                                    > remove_window_internal 0
11:36:34.946  DEBUG  wezterm_gui::termwindow                > PaneOutput: wanted mux_window_id=0 from mux, but was not found, cancel mux subscription
11:36:34.946  DEBUG  mux                                    > removing pane 0
11:36:34.946  DEBUG  wezterm_gui::frontend                  > workspace is default, fixup windows
11:36:34.946  DEBUG  wezterm_gui::frontend                  > workspace is default, fixup windows
11:36:34.943  DEBUG  wezterm_term::terminalstate::performer > perform CSI(Mode(ResetDecPrivateMode(Code(BracketedPaste))))
11:36:34.947  DEBUG  wezterm_term::terminalstate::performer > perform Control(CarriageReturn)
11:36:34.947  DEBUG  wezterm_term::terminalstate::performer > perform Control(CarriageReturn)
11:36:34.947  DEBUG  wezterm_term::terminalstate::performer > perform Control(LineFeed)
11:36:34.947  DEBUG  wezterm_term::terminalstate::performer > perform PrintString("logout")
11:36:34.947  DEBUG  wezterm_term::terminalstate::performer > perform Control(CarriageReturn)
11:36:34.948  DEBUG  wezterm_term::terminalstate::performer > perform Control(LineFeed)
```
Which is where I used ctrl+D to exit out.

### Anything else?

Hope this is somewhat clear! If you believe this to be a Wayland or Kwin bug, just let me know and I'll open a bug report over there.

**Comment by pmoris** [comment]
While browsing through some of the other issues, I found this one: https://github.com/wez/wezterm/issues/5745 (similar trigger, but different outcome).

In any case, after setting `config.front_end = "Software"`, the bug _sometimes_ doesn't appear, approximately 50% of the time.

**Comment by jbrechtel** [comment]
I have a similar effect but it seems to happen sporadically and only after the window has been hidden for some time and it happens when I re-focus it (with its corresponding taskbar shortcut) or click on its taskbar entry.

Often times it will just be resized to be very small (I can see it) and full-screening it via keyboard shortcut will show it again.

I have added the `config.front_end` setting you've mentioned and so far it hasn't recurred - but it wasn't happening 100% of the time to me anyway.

Also on Wayland + Kwin.

No log at the moment.
